Documentation/Relationships.md

```diff
@@ -0,0 +1,165 @@
+# Decoding Relationships
+
+It's very common to have models that relate to other models. When all your
+models conform to `Decodable`, Argo makes it really easy to populate those
+relationships. "How does this work?", you ask. Well, Argo is smart enough to
+know it can decode anything that conforms to `Decodable` because internally,
+Argo is simply calling each type's `decode` function. That means that any type
+that conforms to `Decodable` looks the same to Argo as a `String` or `Int`; in
+fact, Argo can only decode those Swift types because we've already implemented
+`Decodable` for them.
+
+Let's look at a `User`, `Post`, and `Comment` model and how they relate. First,
+our server is sending us the JSON for the `Post` model:
+
+```
+{
+  "author": {
+    "id": 6,
+    "name": "Gob Bluth"
+  },
+  "text": "I've made a huge mistake."
+}
+```
+
+Our `Post` model could then be:
+
+```swift
+struct Post {
+  let author: String
+  let text: String
+}
+```
+
+And, our implementation of `Decodable` for `Post` looks like this:
```

Comment:
Maybe make a note here or earlier that we only care about the user's name at this point, so we'll use the syntax for grabbing the embedded name object.